tests/models/test_modeling_common.py

```diff
@@ -902,6 +915,97 @@ def test_deprecated_kwargs(self):
                 " from `_deprecated_kwargs = [<deprecated_argument>]`"
             )
 
+    @parameterized.expand([True, False])
+    @torch.no_grad()
+    @unittest.skipIf(not is_peft_available(), "Only with PEFT")
+    def test_save_load_lora_adapter(self, use_dora=False):
+        import safetensors
+        from peft import LoraConfig
+        from peft.utils import get_peft_model_state_dict
+
+        from diffusers.loaders.peft import PeftAdapterMixin
+
+        init_dict, inputs_dict = self.prepare_init_args_and_inputs_for_common()
+        model = self.model_class(**init_dict).to(torch_device)
+
+        if not issubclass(model.__class__, PeftAdapterMixin):
+            return
+
+        torch.manual_seed(0)
+        output_no_lora = model(**inputs_dict, return_dict=False)[0]
+
+        denoiser_lora_config = LoraConfig(
+            r=4,
+            lora_alpha=4,
+            target_modules=["to_q", "to_k", "to_v", "to_out.0"],
+            init_lora_weights=False,
+            use_dora=use_dora,
+        )
+        model.add_adapter(denoiser_lora_config)
+        self.assertTrue(check_if_lora_correctly_set(model), "LoRA layers not set correctly")
+
+        torch.manual_seed(0)
+        outputs_with_lora = model(**inputs_dict, return_dict=False)[0]
+
+        self.assertFalse(torch.allclose(output_no_lora, outputs_with_lora, atol=1e-4, rtol=1e-4))
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            model.save_lora_adapter(tmpdir)
+            self.assertTrue(os.path.isfile(os.path.join(tmpdir, "pytorch_lora_weights.safetensors")))
+
+            state_dict_loaded = safetensors.torch.load_file(os.path.join(tmpdir, "pytorch_lora_weights.safetensors"))
+
+            model.unload_lora()
+            self.assertFalse(check_if_lora_correctly_set(model), "LoRA layers not set correctly")
+
+            model.load_lora_adapter(tmpdir, use_safetensors=True)
+            state_dict_retrieved = get_peft_model_state_dict(model, adapter_name="default_0")
+
+            print(f"{state_dict_loaded.keys()=}, {len(state_dict_loaded)}")
+            print(f"{state_dict_retrieved.keys()=} {len(state_dict_retrieved)}")
```

Comment:
@BenjaminBossan do you have an idea as why these two state dicts have different lengths? IMO, they should have the exact same length? Or am I being mistaken here?

The keys that are missing in the retrieved state dict have the `"single_transformer_blocks"` prefix. 

I am on the `peft` main (`0.13.3.dev0`).

Reply:
I debugged and the cause is this line:

https://github.com/huggingface/diffusers/pull/9862/files#diff-fdc2e14a8c1091d917cf31ca155be9baa8095d704bd2a2f58970ec31641c0caeR198

Since `prefix` is defined as `"transformer"` but the keys are starting with `"single_transformer_blocks"`, they don't match. In fact, even the other keys should not match: Even though they start with `"transformer"`, they don't start with `"transformer."` (note the dot), which _I think_ is the intended condition, since the next line replaces `f"{prefix}."`.

I think what happens here is that since this is a standalone module, not a transformer submodule of a larger model, it does not have a `"transformer."`  prefix. Therefore, the keys should not be filtered like that. Probably you know best what the correct solution would be here, but one suggestion is to make the `prefix` argument optional and only filter if it is not `None`.

Also, this problem probably rarely arise in real use cases, as users will typically not load the transformer as a standalone module.

Reply:
Thank you very much!